fe/lib/components/LocationLookup/LocationLookup.tsx

```diff
@@ -0,0 +1,109 @@
+import { ApolloClient, useLazyQuery } from '@apollo/client';
+import {
+  FieldMessage,
+  Loader,
+  Notice,
+  Stack,
+  Text,
+  TextField,
+} from 'braid-design-system';
+import React, { useEffect, useState } from 'react';
+import { InlineCode } from 'scoobie';
+import { useDebounce } from 'use-debounce';
+
+import {
+  LocationQuery,
+  LocationQueryVariables,
+} from '../../types/seekApi.graphql';
+import { flattenResourceByKey } from '../../utils/flattenResourceByKey';
+import { BreadCrumbsString } from '../BreadCrumbsString/BreadCrumbsString';
+import { LOCATION } from '../LocationSuggest/queries';
+import { SeekApiResponse } from '../SeekApiResponse/SeekApiResponse';
+
+export interface LocationLookupProps {
+  schemeId: string;
+  client?: ApolloClient<unknown>;
+  initialLocationId?: string;
+  debounceDelay?: number;
+}
+
+export const LocationLookup = ({
+  schemeId,
+  initialLocationId,
+  debounceDelay = 250,
+  client,
+}: LocationLookupProps) => {
+  const [locationId, setLocationId] = useState(initialLocationId ?? '');
```

Comment:
Do we need a `useEffect` to re-`setLocationId` if `initialLocationId` changes? A scenario I'm thinking of is propagating an ID from query params or local storage to this component, but the ID is (mistakenly?) not available on the first render.

Reply:
Tested locally that `initialLocationId` is set on the first-time rendering. Also we can add the useEffect() function later if any valid scenario need it